### PR TITLE
[Sqlglot] Update datatype_sql for Complex Datatype

### PR DESCRIPTION
**Pull request by robert8138**
This PR updates the datatype_sql method for complex datatype like `ARRAY<TEXT>`, `MAP<TEXT, DOUBLE>`, ... etc. Before this change, all complex data type rendered in DDL are generated with `()` instead of the right struct. 

My approach is to update the DataType expression class to keep track of those delimiters ('(', '<'), and then render them accordingly. @tobymao, let me know if this is reasonable. If not, happy to chat in person to make it better.

### CODE REVIEW

File: tests/test_dialects.py [hunk 1]
```diff
@@ -315,7 +315,7 @@ def test_presto(self):
         )
         self.validate(
             "CAST(ARRAY[1, 2] AS ARRAY<BIGINT>)",
-            "CAST(ARRAY[1, 2] AS ARRAY(BIGINT))",
+            "CAST(ARRAY[1, 2] AS ARRAY<BIGINT>)",
```
Comment (robert8138) [inline tests/test_dialects.py:318]: `ARRAY<TYPE>` and `ARRAY(TYPE)` are both acceptable in Presto.

**Comment by tobymao** [comment]
i wonder instead if we simply detect that it's a nested type, like Struct, Map, and Array, and then generate appropriately

**Comment by robert8138** [comment]
@tobymao, given that you already have a check for whether the datatype is nested or not in the parser, I just use that boolean directly. If nested, we return `data_type<interior>`, otherwise, we return `data_type(interior)`. What do you think?

File: tests/test_dialects.py [hunk 2]
```diff
@@ -297,25 +297,25 @@ def test_presto(self):
...
             "CAST(ARRAY[1, 2] AS ARRAY(BIGINT))",
-            "CAST(ARRAY[1, 2] AS ARRAY(BIGINT))",
+            "CAST(ARRAY[1, 2] AS ARRAY<BIGINT>)",
             read="presto",
             write="presto",
         )
         self.validate(
-            "CAST(MAP(ARRAY[1], ARRAY[1]) AS MAP(ARRAY(INT(9))))",
-            "CAST(MAP(ARRAY[1], ARRAY[1]) AS MAP(ARRAY(INTEGER(9))))",
+            "CAST(MAP(ARRAY[1], ARRAY[1]) AS MAP(INT,INT))",
```
Comment (tobymao) [inline tests/test_dialects.py:311]: why was this test changed

File: sqlglot/generator.py [hunk 3]
```diff
@@ -305,9 +305,15 @@ def cte_sql(self, expression):
     def datatype_sql(self, expression):
         type_value = expression.this
         type_sql = self.type_mapping.get(type_value, type_value.value)
-        args = self.expressions(expression, flat=True)
-        args = f"({args})" if args else ""
-        return f"{type_sql}{args}"
+        interior = self.expressions(expression, flat=True)
+        if interior:
```
Comment (tobymao) [inline sqlglot/generator.py:309]: nested = ""
if interirior:
  nested = "<{interior}>" if expression.args["nested"] else "({interior})"

File: tests/test_dialects.py (hunk 2 above)
Comment (robert8138) [reply tests/test_dialects.py:311]: `CAST(MAP(ARRAY[1], ARRAY[1]) AS MAP(ARRAY(INT(9))))` is not a valid Presto query (I think?), so I updated it. But maybe you were trying to test double nested structure like `MAP(ARRAY(INT(9)))`

File: sqlglot/generator.py (hunk 3 above)
Comment (robert8138) [reply sqlglot/generator.py:310]: Fixed

File: tests/test_dialects.py (hunk 2 above)
Comment (robert8138) [reply tests/test_dialects.py:311]: Makes sense, added another test with double nested structure.

**Comment by robert8138** [comment]
Addressed the comments, PTAL again. Thank you.

**Pull request merged**